tools/gasmonitor/integration_test.go

```diff
@@ -0,0 +1,270 @@
+				msgSend1 := banktypes.NewMsgSend(
+					getAddress(account1, s.cctx.Keyring),
+					getAddress(account2, s.cctx.Keyring),
+					sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(1))),
+				)
+				account3 := s.unusedAccount()
+				msgSend2 := banktypes.NewMsgSend(
+					getAddress(account1, s.cctx.Keyring),
+					getAddress(account3, s.cctx.Keyring),
+					sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(1))),
+				)
+				return []sdk.Msg{msgSend1, msgSend2}, account1
+			},
+			expectedCode: abci.CodeTypeOK,
+		},
+	}
+
+	// sign and submit the transactions
+	for i, tt := range tests {
+		msgs, signer := tt.msgFunc()
+		res, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, signer, msgs...)
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		assert.Equal(t, abci.CodeTypeOK, res.Code, tt.name)
+		tests[i].hash = res.TxHash
+	}
+
+	require.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	for _, tt := range tests {
+		res, err := testnode.QueryTx(s.cctx.Context, tt.hash, true)
+		assert.NoError(t, err)
+		assert.Equal(t, tt.expectedCode, res.TxResult.Code, tt.name)
+	}
+
+	// run the PFBs after these tests have ran to increment the sequence numbers
+	// for the above accounts
+	s.submitPayForBlob()
+}
+
+func (s *GasConsumptionTestSuite) submitPayForBlob() {
+	t := s.T()
+
+	type test struct {
+		name  string
+		blobs []*blobtypes.Blob
+		opts  []blobtypes.TxBuilderOption
+	}
+
+	tests := []test{
+		{
+			"small random typical",
+			blobfactory.ManyRandBlobs(t, tmrand.NewRand(), 1),
+			[]blobtypes.TxBuilderOption{
+				blobtypes.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(9000)))),
+				blobtypes.SetGasLimit(90_000),
+			},
+		},
+		{
+			"small random typical",
+			blobfactory.ManyRandBlobs(t, tmrand.NewRand(), 1, 1),
+			[]blobtypes.TxBuilderOption{
+				blobtypes.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(9000)))),
+				blobtypes.SetGasLimit(90_000),
+			},
+		},
+		{
+			"small random with memo",
+			blobfactory.ManyRandBlobs(t, tmrand.NewRand(), 1),
+			[]blobtypes.TxBuilderOption{
+				blobtypes.SetMemo("lol I could stick the rollup block here if I wanted to"),
+				blobtypes.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(9000)))),
+				blobtypes.SetGasLimit(90_000),
+			},
+		},
+		{
+			"large random typical",
+			blobfactory.ManyRandBlobs(t, tmrand.NewRand(), 100_000),
+			[]blobtypes.TxBuilderOption{
+				blobtypes.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(100000)))),
+				blobtypes.SetGasLimit(1_000_000),
+			},
+		},
+	}
+	count := 0
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			signer := blobtypes.NewKeyringSigner(s.cctx.Keyring, s.accounts[count], s.cctx.ChainID)
+			res, err := blob.SubmitPayForBlob(context.TODO(), signer, s.cctx.GRPCClient, tc.blobs, tc.opts...)
+			require.NoError(t, err)
+			require.NotNil(t, res)
+			require.Equal(t, abci.CodeTypeOK, res.Code, res.Codespace, res.Logs)
+		})
+		count++
+	}
```

Comment:
[optional refactor] Is it possible to remove the `count` variable entirely?

```suggestion
	for i, tc := range tests {
		s.Run(tc.name, func() {
			signer := blobtypes.NewKeyringSigner(s.cctx.Keyring, s.accounts[i], s.cctx.ChainID)
			res, err := blob.SubmitPayForBlob(context.TODO(), signer, s.cctx.GRPCClient, tc.blobs, tc.opts...)
			require.NoError(t, err)
			require.NotNil(t, res)
			require.Equal(t, abci.CodeTypeOK, res.Code, res.Codespace, res.Logs)
		})
	}
```

Reply:
yeah good catch forgot about that lil guy